mx.h

```diff
@@ -212,13 +250,20 @@ struct MxOps
    * @param msg Message to close
    * @retval  0 Success
    * @retval -1 Failure
+   *
+   * **Contract**
+   * - @a m   is not NULL
+   * - @a msg is not NULL
    */
   int (*msg_close)       (struct Mailbox *m, struct Message *msg);
 
   /**
    * msg_padding_size - Bytes of padding between messages
    * @param m Mailbox
    * @retval num Bytes of padding
+   *
+   * **Contract**
+   * - @a m is not NULL
    */
   int (*msg_padding_size)(struct Mailbox *m);
 
```

Comment:
No contract for `msg_save_hcache()`, below

Reply:
yes, as the function as is didn't have any guarantees.
It only checks if it has the mx_ops and then calls the worker function.
in fact, the only thing guaranteed right now is that the mailbox isn't null and only because the application would blow up if it was ;)

Shall I add the contract (and the guards)?